File: Week 02/id_682/242.valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if  len(s) != len(t):
            return False
        
        res = {}
        for i in s:
            if i in res:
                res[i] += 1
            else:
                res[i] = 1
        
        for j in t:
            if j in res:
                res[j] -= 1
            else:
                return False
        
        for check in res:
            if res[check] != 0:
                return False
        return True

File: Week 02/id_682/242.valid-anagram.php
class Solution {

    /**
     * @param String $s
     * @param String $t
     * @return Boolean
     */
    function isAnagram($s, $t) {
        if (strlen($s) != strlen($t)) {
            return false;
        }
        
        $res = [];
        for ($i = 0; $i < strlen($s); $i++) {
            if (isset($res[$s[$i]])) {
                $res[$s[$i]]++;
            } else {
                $res[$s[$i]] = 1;
            }
            
            if (isset($res[$t[$i]])) {
                $res[$t[$i]]--;
            } else {
                $res[$t[$i]] = -1;
            }
        }
        
        foreach ($res as $val) {
            if ($val != 0) {
                return false;
            }
        }
        
        return true;
    }
}